app/lib/server/functions/saveUser.js

```diff
@@ -146,8 +147,13 @@ function validateUserEditing(userId, userData) {
 
 	const canEditOtherUserInfo = hasPermission(userId, 'edit-other-user-info');
 	const canEditOtherUserPassword = hasPermission(userId, 'edit-other-user-password');
+	const user = Users.findOneById(userData._id);
+	const editingUserRoles = userData.roles
+							&& user.roles.length === userData.roles.length
+							&& user.roles.every((role, index) => role === user[index]);
+	const editingField = (previousValue, newValue) => newValue && newValue !== previousValue;
```

Comment:
checking if newValue exists wouldn't introduce an edge case like allowing to delete a custom status even though the edit flag is off?

Reply:
Ohh, it would. I've just removed this check, think we're not missing anything by removing it. Tks 🤗